microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -163,6 +171,60 @@ func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient
 					}
 				}
 
+				if createdObj.GetKind() == "Deployment" {
+					var newEnvVars []corev1.EnvVar
+					if svcCfg != nil {
+						for name, value := range *svcCfg {
+							if name == "endpoint" {
+								continue
+							}
+							itemEnvVar := corev1.EnvVar{
+								Name:  name,
+								Value: value,
+							}
+							newEnvVars = append(newEnvVars, itemEnvVar)
+						}
+					}
+					if len(newEnvVars) > 0 {
+						deployment := &appsv1.Deployment{}
+						err = runtime.DefaultUnstructuredConverter.FromUnstructured(createdObj.UnstructuredContent(), deployment)
+						if err != nil {
+							fmt.Printf("Failed to save deployment: %v\n", err)
+						}
+						for i := range deployment.Spec.Template.Spec.Containers {
+							deployment.Spec.Template.Spec.Containers[i].Env = append(
+								deployment.Spec.Template.Spec.Containers[i].Env,
+								newEnvVars...)
+						}
+						modifiedObj, derr := runtime.DefaultUnstructuredConverter.ToUnstructured(deployment)
+						if derr != nil {
+							fmt.Printf("Failed to marshal updated deployment: %v", derr)
+						}
+
+						// Remove managedFields from the unstructured object
+						if _, ok := modifiedObj["metadata"].(map[string]interface{}); ok {
+							delete(modifiedObj["metadata"].(map[string]interface{}), "managedFields")
+						}
+
+						modifiedUnstructured := &unstructured.Unstructured{Object: modifiedObj}
+						modifiedBytes, merr := json.Marshal(modifiedUnstructured)
+						if merr != nil {
+							fmt.Printf("Failed to marshal updated deployment: %v", merr)
+						}
+						gvr := appsv1.SchemeGroupVersion.WithResource("deployments")
```

Comment:
is it possible to call the function applyResourceToK8s here?

Reply:
Hi @KfreeZ, I suggest not, because I use `runtime.DefaultUnstructuredConverter.ToUnstructured` to convert a deployment, and `yaml.NewDecodingSerializer` operation in `applyResourceToK8s` which is unnecessary.